pkg/ottl/parser_collection.go

```diff
@@ -0,0 +1,289 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottl // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+
+import (
+	"fmt"
+	"reflect"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+var _ interface {
+	ParseStatements(statements []string) ([]*Statement[any], error)
+} = (*Parser[any])(nil)
+
+var _ ParsedStatementConverter[any, StatementsGetter, any] = func(
+	_ *ParserCollection[StatementsGetter, any],
+	_ *Parser[any],
+	_ string,
+	_ StatementsGetter,
+	_ []*Statement[any]) (any, error) {
+	return nil, nil
+}
+
+// StatementsGetter represents a set of statements to be parsed
+type StatementsGetter interface {
+	// GetStatements retrieves the OTTL statements to be parsed
+	GetStatements() []string
+}
+
+// ottlParserWrapper wraps an ottl.Parser using reflection, so it can invoke exported
+// methods without knowing its generic type (transform context).
+type ottlParserWrapper[S StatementsGetter] struct {
+	parser                         reflect.Value
+	prependContextToStatementPaths func(context string, statement string) (string, error)
+}
```

Comment:
Does this type need a generic? I don't see where it uses it.

Reply:
Good catch! that's correct, it does not need to be generic.